admin-dev/themes/new-theme/js/pages/order/create/product-renderer.js

```diff
@@ -252,17 +248,29 @@ export default class ProductRenderer {
 
     const templateTypeMap = {
       [fieldTypeFile]: $fileInputTemplate,
-      [fieldTypeText]: $textInputTemplate,
+      [fieldTypeText]: $textInputTemplate
     };
 
     for (const key in customizationFields) {
       const customField = customizationFields[key];
       const $template = templateTypeMap[customField.type].clone();
 
-      $template.find(createOrderMap.productCustomInput)
+      if (customField.type === fieldTypeFile) {
+        $template.on('change', e => {
+          const fileName = e.target.files[0].name;
+
+          $(e.target)
+            .next('.custom-file-label')
+            .html(fileName);
+        });
+      }
```

Comment:
This is the modification, others is linting/correcting, should have pushed on another commit, sorry for the reviewers